freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/KerberosMgmt/KerberosMgmtV1Endpoint.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.freeipa.api.v1.KerberosMgmt;
+
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabModelNotes;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabOperationsDescription;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabRequest;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabResponse;
+import com.sequenceiq.service.api.doc.ContentType;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Path("/v1/kerberosmgmt")
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/v1/kerberosmgmt", protocols = "http,https")
+public interface KerberosMgmtV1Endpoint {
+
+    @POST
+    @Path("createprincipal")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = KeytabOperationsDescription.DESCRIBE_CREATE_HOST_PRINCIPAL, produces = ContentType.JSON,
+            notes = KeytabModelNotes.CREATE_HOST_AND_PRINCIPAL_NOTES,
+            nickname = "getServiceKeytabV1")
+    void createHostAndPrincipal(@Valid ServiceKeytabRequest request);
```

Comment:
Based on my prior comments, I think this should return a keytab.

Reply:
done.